server/src/main/java/org/opensearch/cluster/coordination/LeaderChecker.java

```diff
@@ -158,6 +162,15 @@ public void onNodeDisconnected(DiscoveryNode node, Transport.Connection connecti
                 handleDisconnectedNode(node);
             }
         });
+        initializeMetrics(metricsRegistry);
+    }
+
+    private void initializeMetrics(MetricsRegistry metricsRegistry) {
+        this.leaderCheckFailureCounter = metricsRegistry.createCounter(
```

Comment:
Even for other metrics we are adding them inline only (ref: https://github.com/opensearch-project/OpenSearch/pull/12333) and not moving to a separate class. Moreover, I believe creating new classes for each set of metrics can lead to a lot of new classes where we will have to provide getters and setters for each of the metric separately. 
Still, if you feel we should move them to new classes, please let me know.

Reply:
The dependency on MetricsRegistry and its associated methods (createCounter, createTimer) has to be taken in all the classes which needs instrumentation. 

Rather a Counter class could set-up counters/metrics (along with the definition). This can then be referenced by instrumented class. I see something similar for Search related metrics :- SearchQueryCounters (we might require a static version of the same) . Thoughts ?